spyderlib/spyder.py

```diff
@@ -1928,6 +1940,56 @@ def quick_layout_switch(self, index):
             action = plugin.toggle_view_action
             action.setChecked(plugin.dockwidget.isVisible())
 
+    # --- Other
+    def save_visible_toolbars(self):
+        """ """
+        print('save')
+        temp = []
+        for toolbar in self.visible_toolbars:
+            temp.append(toolbar.objectName())
+        CONF.set('main', 'last_visible_toolbars', temp)
+
+    def load_last_visible_toolbars(self):
+        """ """
+        print('load')
+        toolbars_names = CONF.get('main', 'last_visible_toolbars')
+        print(toolbars_names, type(toolbars_names))
+        if not toolbars_names:
+            self.get_visible_toolbars()
+        else:
+            dic = {}
+            for toolbar in self.toolbarslist:
+                dic[toolbar.objectName()] = toolbar
+    
+            self.visible_toolbars = []            
+            toolbars_names = CONF.get('main', 'last_visible_toolbars')
+            for name in toolbars_names:
+                if name in dic:
+                    self.visible_toolbars.append(dic[name])
+        print(self.visible_toolbars)
+
+    def get_visible_toolbars(self):
+        self.visible_toolbars = []

```

Comment:
I think it's better to use a temporary variable here, instead of assigning to the attribute directly, and at the end of this method assign `self.visible_toolbars` the value of that temp var.